lib/tasks/compact_index.rake

```diff
@@ -1,18 +1,41 @@
 namespace :compact_index do
   desc "Fill yanked_at with current time"
-  task migrate_yanked_at: :environment do
-    Version.where(yanked_at: nil).find_in_batches do |versions|
-      versions.each { |v| v.update_attribute :yanked_at, Time.now.utc }
+  task backfill_yanked_at: :environment do
+    without_yanked_at = Version.where(yanked_at: nil)
+    mod = ENV['shard']
+    without_yanked_at = without_yanked_at.where("id % 4 = ?", mod.to_i) if mod
+
+    total = without_yanked_at.count
+    i = 0
+    puts "Total: #{total}"
+    without_yanked_at.find_each do |version|
+      version.update_attribute :yanked_at, Time.now.utc
```

Comment:
Yeah, I wrote this originally for bundler-api, where we don't hold the deletions. If you can extract the right information for filling this column, the better. I don't think it needs to be equal from the bundler_api. It is used to mount the `/versions` answer, and it is not important to have the correct date to work properly as long as the versions file is generated after `Time.now.utc`. It will be great if you can fill this column with the correct information, or better yet remove this column and grab the correct information from the deletions table. It is only used [here](https://github.com/rubygems/rubygems.org/blob/master/app/models/rubygem.rb#L267).

@segiddins do you agree with me? Can we just drop the `yanked_at` column on rubygems if we can have this information from other table we don't have on bundler-api?